### DISCUSSION TITLE
When attempting to synchronize with a Git repository (we are using a corporate GitLab instance), the following error message appears:

### DISCUSSION THREAD

**Discussion by it-openspace-shamil** [netbox-community/netbox#17200] [Q&A]
Version: Netbox 4.0.9

Issue: When attempting to synchronize with a Git repository (we are using a corporate GitLab instance), the following error message appears:

UnboundLocalError("local variable '_' referenced before assignment")

Details:

I verified that a curl request to the site works fine.
The repository was created without requiring authentication (it's public), though the same error occurs when authentication is enabled.
I was able to connect and synchronize with the same Git server in other projects (e.g., Semaphore) without any issues.
Could you please help resolve this problem?

**Comment by candlerb** [comment]
What exactly are you doing to trigger the error message? How is this error message seen, e.g. is it within the Netbox GUI, or returned from the REST API, or something else?

If it's within the GUI, set DEBUG=True in configuration.py (or ADMINS and EMAIL) to obtain a full backtrace.

**Comment by markkuleinio** [comment]
Search for "UnboundLocalError" in this repo and see that a previous reason for this error is that you haven't installed `dulwich` package.

**Comment by it-openspace-shamil** [comment]
Thanks))
